src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    const currentAnimal = Animal.alive.find(animal => animal === herbivore);
+
+    if (currentAnimal.hidden !== undefined && currentAnimal.hidden !== true) {
+      currentAnimal.health -= 50;
+    }
+
+    if (currentAnimal.health <= 0) {
+      Animal.alive = Animal.alive.filter(animal => animal !== currentAnimal);
+    }
```

Comment:
Ok, but object comparing in this case looks strange, as for me better to filter animals by `health` property